Dart: Document Scope Deprecations (#5145)
getsentry/sentry-docs: src/platforms/dart/migration.mdx

}
```

> - `Scope.user` setter was deprecated in favor of `Scope.setUser` and it will be removed in a future update.

Reviewer: ```suggestion
- The `Scope.user` setter was deprecated in favor of `Scope.setUser`, and it will be removed in a future update.
```

> - 'Scope.user' setter was deprecated in favour of `Scope.setUser` and it will be removed in a future update.

Reviewer: ```suggestion
- `Scope.user` setter was deprecated in favor of `Scope.setUser` and it will be removed in a future update.
```

Reviewer: I guess `favor` and `favour` are correct, let's @imatwawana help us out :)

Reviewer 2: Favor - we use American spellings in the docs

> - 'Scope.user' setter was deprecated in favour of `Scope.setUser` and it will be removed in a future update.

Reviewer 2: ```suggestion
- The 'Scope.user' setter was deprecated in favor of `Scope.setUser`, and it will be removed in a future update.
```

> - `Scope.attachements` getter was deprecated in favour of `attachments` and it will be removed in a future update.

Reviewer: ```suggestion
- `Scope.attachements` getter was deprecated in favor of `attachments` and it will be removed in a future update.
```

> - `Scope.attachements` getter was deprecated in favor of `attachments` and it will be removed in a future update.

Reviewer: ```suggestion
- The `Scope.attachements` getter was deprecated in favor of `attachments`, and it will be removed in a future update.
```

- The `Scope.attachements` getter was deprecated in favor of `attachments`, and it will be removed in a future update.

> - The following `Scope` methods now return `Future<void>` instead od `void`:

Reviewer: ```suggestion
- The following `Scope` methods now return `Future<void>` instead of `void`:
```
